experiments/utils.py

```diff
@@ -366,8 +375,19 @@ def _get_disabled_experiment_names(self):
         )
 
     def set_disabled_experiments(self, names):
-        Enrollment.objects.filter(experiment__name__in=names).update(disabled=True)
-        Enrollment.objects.exclude(experiment__name__in=names).update(disabled=False)
+        for name in names:
+            experiment = experiment_manager.get_experiment(name, auto_create=False)
+            if not experiment:
+                continue
+            enrollment, _ = Enrollment.objects.get_or_create(
+                experiment=experiment,
+                user=self.user,
+                defaults={'disabled': True}
+            )
+            if not enrollment.disabled:
+                enrollment.disabled = True
+                enrollment.save()
```

Comment:
do we need this block? if we replace the block above with `update_or_create` we can remove this block, right?

Reply:
also you are creating enrollments for the user with all experiments, shouldn't the user be enrolled in the experiment only if he/she reaches some page? that will influence the results, they enrolled and didn't reach a goal...

Reply:
True, now using `update_or_create`. I often forget that it exists...

Re creating enrollments - yes but the user is not enrolled into any particular alternative here , so `_get_enrollment` will return `None` as it should. The enrollment only holds the information that the experiment is disabled.

And in the other case, when the user has already been enrolled into the experiment on another (non-disabled) page, then here we only switch the `disabled` flag to `True`. If at a later time the flag is switched back to `disabled=False` the alternative will become visible again, so to speak, i.e. `_get_enrollment` will start returning that value once again.